src/main/io/osd.c

```diff
@@ -425,19 +428,34 @@ static void osdDrawSingleElement(uint8_t item)
 #endif // GPS
 
     case OSD_COMPASS_BAR:
-        {
-            int16_t h = DECIDEGREES_TO_DEGREES(attitude.values.yaw);
-
-            h = osdGetHeadingIntoDiscreteDirections(h, 16);
+    {
+#ifdef USE_GPS
+        int16_t h = 0;
+		// If MAG_Heading is not available, yaw starts with zero in a random direction. With GPS course over ground the flight direction is roughly displayed instead
+		if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME) && !FLIGHT_MODE(MAG_MODE)) {
+			h = DECIDEGREES_TO_DEGREES(gpsSol.groundCourse);
```

Comment:
This is wrong, as it will take 0 for the heading if there is no GPS fix.

Reply:
Thanks - fixed.
Now the MAG heading is only overridden if GPS is available but no MAG.